client/my-sites/hosting/phpmyadmin-card/restore-db-password.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { getHttpData, httpData, requestHttpData } from 'state/data-layer/http-data';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import Dialog from 'components/dialog';
+import { connect } from 'react-redux';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { successNotice } from 'state/notices/actions';
+
+const requestId = siteId => `restore-db-password-${ siteId }`;
+
+export const restoreDatabasePassword = siteId =>
+	requestHttpData(
+		requestId( siteId ),
+		http(
+			{
+				method: 'POST',
+				path: `/sites/${ siteId }/hosting/restore-database-password`,
+				apiNamespace: 'wpcom/v2',
+				body: {},
+			},
+			{}
+		),
+		{
+			fromApi: () => () => [ [ requestId( siteId ), null ] ],
+			freshness: 0,
+		}
+	);
+
+const RestorePasswordDialog = ( {
+	isVisible,
+	onRestore,
+	onCancel,
+	isRestoring,
+	hasRestored,
+	siteId,
+	translate,
+	showSuccessNotice,
+	reset,
+} ) => {
+	if ( hasRestored ) {
+		reset();
```

Comment:
What happens if we don't reset?

Reply:
The request will remain in `httpData` as succeeded, keeping `hasRestored` as true. So any re-render (i.e. by clicking on the restore password link again) will automatically fire a success notice and hide the dialog.